quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/web/QuarkusRequestFilter.java

```diff
@@ -76,18 +76,29 @@ private boolean ignoreContext(RoutingContext context) {
 
     private Runnable createBlockingHandler(RoutingContext context) {
         return () -> {
-            KeycloakSession session = configureContextualData(context);
+            configureContextualData(context);
 
             try {
-                context.next();
+                context.addBodyEndHandler(event -> close(context));
```

Comment:
You are introducing some of the issues fixed here https://github.com/keycloak/keycloak/pull/15193.

Also, this method is not supposed to be used for resource cleanup.

FYI, most of the time the session is closed from the response filter/interceptors as they run within the same worker thread and right after the response is created from JAX-RS resource methods.

Closing the session here in this top-level filter is to make sure the session is actually closed (we can easily check for "session leaks" here as well plus avoid dispatching requests if they can be served directly from an event loop thread) as well as make sure the session is closed when Resteasy handlers are not executed.

Differently from Resteasy Reactive, Resteasy Classic does not provide a single point to hook all the logic to manage the request lifecycle. That is about to change once we move to Resteasy Reactive.